icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -240,6 +241,13 @@ void U_CALLCONV initUnitExtras(UErrorCode& status) {
     TableKeysSink identifierSink(gSimpleUnits, simpleUnitsCount);
     ures_getAllItemsWithFallback(unitsBundle.getAlias(), "convertUnits", identifierSink, status);
     for (int i = 0; i < simpleUnitsCount; i++) {
+        if (uprv_strcmp(gSimpleUnits[i], "kilogram") == 0) {
```

Comment:
I have a slight preference for putting this check inside of the resource sink (TableKeysSink), such that `"kilogram"` doesn't get added to gSimpleUnits in the first place.

Reply:
"TableKeysSink" being conceived of as generic and reusable ("just collect the keys from a table"), this would be making it more "custom" (and less likely to be promoted as a possibly shared utility). I'll think on this a bit longer - it would at least need a new narrower name.